src/dock_state/tree/node.rs

```diff
@@ -288,4 +288,35 @@ impl<Tab> Node<Tab> {
             _ => Default::default(),
         }
     }
+
+    /// Returns a new Node while mapping the tab type
+    pub fn map_tabs<F, NewTab>(&self, function: F) -> Node<NewTab>
+    where
+        F: FnMut(&Tab) -> NewTab,
+    {
+        match self {
```

Comment:
This whole `match` can be simplified a lot, as almost all it does is return a copy of the node and sometimes override one field.

Reply:
Actually nevermind, the returned `Node` has a different type.